eslint.config.mjs

```diff
@@ -0,0 +1,403 @@
+// @ts-check
+
+import js from '@eslint/js';
+import { globalIgnores } from 'eslint/config';
+import formatjs from 'eslint-plugin-formatjs';
+// @ts-expect-error -- No typings
+import importPlugin from 'eslint-plugin-import';
+import jsdoc from 'eslint-plugin-jsdoc';
+import jsxA11Y from 'eslint-plugin-jsx-a11y';
+import promisePlugin from 'eslint-plugin-promise';
+import react from 'eslint-plugin-react';
+import reactHooks from 'eslint-plugin-react-hooks';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+
+/** @type {import('typescript-eslint').ConfigArray} */
+export const baseConfig = [
+  js.configs.recommended,
+  importPlugin.flatConfigs.recommended,
+];
+
+export default tseslint.config([
+  baseConfig,
+  globalIgnores([
+    'build/**/*',
+    'coverage/**/*',
+    'db/**/*',
+    'lib/**/*',
+    'log/**/*',
+    'node_modules/**/*',
+    'public/**/*',
+    '!public/embed.js',
+    'spec/**/*',
+    'tmp/**/*',
+    'vendor/**/*',
+    'streaming/**/*',
+  ]),
+  react.configs.flat.recommended,
+  react.configs.flat['jsx-runtime'],
+  reactHooks.configs['recommended-latest'],
+  jsxA11Y.flatConfigs.recommended,
+  importPlugin.flatConfigs.react,
+  promisePlugin.configs['flat/recommended'],
+  // @ts-expect-error -- For some reason the formatjs package exports an empty object?
+  formatjs.configs.strict,
+  jsdoc.configs['flat/recommended'],
+  {
+    linterOptions: {
+      reportUnusedDisableDirectives: 'error',
+      reportUnusedInlineConfigs: 'error',
+    },
+
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+      },
+
+      parser: tseslint.parser,
+      ecmaVersion: 2021,
+      sourceType: 'module',
+    },
+
+    settings: {
+      react: {
+        version: 'detect',
+      },
+
+      'import/ignore': ['node_modules', '\\.(css|scss|json)$'],
+
+      'import/resolver': {
+        typescript: {},
+      },
+    },
+
+    rules: {
+      'consistent-return': 'error',
```

Comment:
We should probably have some of those rules in a `baseConfig`, as we want them to apply in all our projects?
At least those that are not
- React/browser specific
- Not ones we override for typescript

Reply:
done with 8d85cb3